--- a/manager/pkg_observer.c	2026-09-04 22:51:11.152063995 -0300
+++ b/manager/pkg_observer.c	2026-09-04 22:51:11.167603839 -0300
@@ -21,6 +21,7 @@
 
 static struct fsnotify_group *g;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0)
 static int ksu_handle_inode_event(struct fsnotify_mark *mark, u32 mask,
                                   struct inode *inode, struct inode *dir,
                                   const struct qstr *file_name, u32 cookie)
@@ -35,9 +36,31 @@
     }
     return 0;
 }
+#else
+static int ksu_handle_event(struct fsnotify_group *group,
+                            struct inode *inode, u32 mask,
+                            const void *data, int data_type,
+                            const unsigned char *file_name, u32 cookie,
+                            struct fsnotify_iter_info *iter_info)
+{
+    if (!file_name)
+        return 0;
+    if (mask & FS_ISDIR)
+        return 0;
+    if (strlen(file_name) == 13 && !memcmp(file_name, "packages.list", 13)) {
+        pr_info("packages.list detected: %d\n", mask);
+        track_throne(false);
+    }
+    return 0;
+}
+#endif
 
 static const struct fsnotify_ops ksu_ops = {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 1, 0)
 	.handle_inode_event = ksu_handle_inode_event,
+#else
+	.handle_event = ksu_handle_event,
+#endif
 };
 
 static int add_mark_on_inode(struct inode *inode, u32 mask,
